CRM/Core/Payment/Stripe.php

```diff
@@ -669,4 +669,23 @@ function doRecurPayment(&$params, $amount, $stripe_customer) {
   function doTransferCheckout(&$params, $component) {
     CRM_Core_Error::fatal(ts('Use direct billing instead of Transfer method.'));
   }
+
+  /**
+   * Stripe payment instrument validation.
+   *
+   * @param array $values
+   * @param array $errors
+   */
+  public function validatePaymentInstrument($values, &$errors) {
+    $mandatoryFields = $this->getMandatoryFields();
+
+    // set credit_card_number and cvv2 as not required as we have stripe_token
+    $mandatoryFields['credit_card_number']['is_required'] = FALSE;
+    $mandatoryFields['cvv2']['is_required'] = FALSE;
+    CRM_Core_Form::validateMandatoryFields($mandatoryFields, $values, $errors);
+
+    if (empty($_POST['stripe_token'])) {
```

Comment:
I guess if token is not passed there is no need to perform other validations? So would it be better to exec this check earlier in the function and return?

```php
if (empty($_POST['stripe_token'])) {
  $errors['stripe_token'] = ts('Stripe token not found.');
  
  return;
}

$mandatoryFields = $this->getMandatoryFields();
// and the rest of code...
```

Also, it seems a bit weird that we are accessing $_POST inside this func, isn't token passed inside `$values`? Just checking.